src/main/java/com/github/crimsondawn45/fabricshieldlib/mixin/LivingEntityMixin.java

```diff
@@ -19,40 +21,80 @@
  */
 @SuppressWarnings("ALL")
 @Mixin(LivingEntity.class)
-public class LivingEntityMixin {
-
-    @Inject(at = @At(value = "HEAD"), method = "damage(Lnet/minecraft/entity/damage/DamageSource;F)Z")
-    private void invokeEvent(DamageSource source, float amount, CallbackInfoReturnable<Boolean> callbackInfo) {
-        // Handle shield blocking
-        LivingEntity entity = (LivingEntity) (Object) this;
-        ItemStack activeItem = entity.getActiveItem();
-        ShieldBlockCallback.EVENT.invoker().block(entity, source, amount, entity.getActiveHand(), activeItem);
-    }
-
-    @Inject(at = @At(value = "TAIL"), method = "damage(Lnet/minecraft/entity/damage/DamageSource;F)Z")
```

Comment:
Could you explain why this was unnecessary? I remember it used to be one big mixin which we then split up into head and tail.

Reply:
The previous implementation requires a lot of duplication of the vanilla function. Mixin itself cannot manipulate branching instruction well (`if (item instanceof ShieldItem`), that was why we had to duplicate some vanilla code at the first place.

Now, we took advantage of MixinExtra's ModifyExpressionValue, which allows us to manipulate the result of `item instanceof ShieldItem`. The whole chunk of duplicated codes can now be removed.

Furthermore, the addition of DataComponents somehow reshaped the vanilla shield implementation and reduced the amount of our work.